addons/dialogue_nodes/editor/graph.gd

```diff
@@ -151,6 +152,9 @@ func connect_node_signals(node : GraphNode):
 			characters_updated.connect(node._on_characters_updated)
 			node.disconnection_from_request.connect(_on_disconnection_from_request)
 			node.connection_shift_request.connect(_on_connection_shift_request)
+		6: # fork node
```

Comment:
Is this required?

Reply:
I think so. I used your DialogueNode as reference to build the Fork, and these signals are used to connect/disconnect stuff through Option editing. In that regard, the Fork and Dialogue Nodes work pretty similarly. Removing these signal connections makes Node connections desync whenever you add/remove options from the fork.

Reply:
Maybe its one of those things that could be done _within_ the Node code, rather than doing it from the graph by keeping a reference of the graph in the Node, rather than emit signals to it.

`graph_ref = graph; graph_ref.disconnect_node(yadayada)`, instead of `graph.connect.node_disconnect_request(data)`.

Reply:
I guess making the Godot nodes communicate through signals is the recommended way to go. Lets keep this code as is. I totally forgot about the fact that DialogueNode has connections shift when options are changed.